internal/backend/service.go

```diff
@@ -29,11 +84,114 @@ func NewDefaultService(c Persister, l log.Logger) *DefaultService {
 	}
 }
 
-// GetAccessRequest will retrieve the access request from k8s identified by the
-// given name and namespace. Will return a nil value without any error if the
-// access request isn't found.
-func (s *DefaultService) GetAccessRequest(ctx context.Context, name, namespace string) (*api.AccessRequest, error) {
-	ar, err := s.k8s.GetAccessRequest(ctx, name, namespace)
+// GetAccessRequest will retrieve the access request for the specified role.
+// Will return a nil value without any error if an access request isn't found for this role.
+func (s *DefaultService) GetAccessRequest(ctx context.Context, key *AccessRequestKey, roleName string) (*api.AccessRequest, error) {
+
+	// get all access requests
+	accessRequests, err := s.ListAccessRequests(ctx, key, true)
+	if err != nil {
+		return nil, fmt.Errorf("error listing access request for role %s: %w", roleName, err)
+	}
+
+	// find the first access request matching the requested role
+	for _, ar := range accessRequests {
+		if ar.Spec.Role.TemplateName == roleName {
+			return ar, nil
+		}
+	}
+
+	return nil, nil
+}
+
+func (s *DefaultService) ListAccessRequests(ctx context.Context, key *AccessRequestKey, shouldSort bool) ([]*api.AccessRequest, error) {
+	accessRequests, err := s.k8s.ListAccessRequests(ctx, key)
+	if err != nil {
+		return nil, fmt.Errorf("error getting accessrequest from k8s: %w", err)
+	}
+
+	filtered := []*api.AccessRequest{}
+	for i, ar := range accessRequests.Items {
+		if ar.Status.RequestState == api.ExpiredStatus {
```

Comment:
`InvalidStatus` does not yet exist. When it is added, the businees logic should be updated because it should not be filtered if it is the only item so the client can have feedback on that request being invalid.

Reply:
Sure. Please add a TODO comment in this case.